airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -138,15 +141,39 @@ def get_conn(self) -> Connection:
             )
         return self._sql_conn
 
+    @overload
+    def run(
+        self,
+        sql: str | Iterable[str],
+        autocommit: bool = False,
+        parameters: Iterable | Mapping | None = None,
+        handler: None = None,
+        split_statements: bool = True,
+        return_last: bool = True,
+    ) -> None:
+        ...
+
+    @overload
+    def run(
+        self,
+        sql: str | Iterable[str],
+        autocommit: bool = False,
+        parameters: Iterable | Mapping | None = None,
+        handler: Callable[[Any], T] = None,  # type: ignore[assignment]
+        split_statements: bool = True,
+        return_last: bool = True,
+    ) -> T | list[T]:
```

Comment:
I think this can be further split into `Literal[True]` returning `T` and `Literal[False]` returning `list[T]`

Reply:
Yes, you're correct. I decided initially that I wanted to take baby steps with this PR, but we can go all the way if you'd like!

Reply:
Do whatever you feel comfortable